dispatches/models/nuclear_case/flowsheets/nuclear_flowsheet_multiperiod_class.py

```diff
@@ -0,0 +1,268 @@
+        """
+        return value(b.P_T[last_implemented_time_step])
+
+    @staticmethod
+    def get_implemented_profile(b, last_implemented_time_step):
+
+        """
+        This method gets the implemented variable profiles in the last optimization solve.
+
+        Args:
+            b: a Pyomo block
+            last_implemented_time_step: time index for the last implemented time step
+
+         Returns:
+             profile: the intended profile, {unit: [...]}
+        """
+        active_blks = b.nuclear.get_active_process_blocks()
+        implemented_tank_holdup = deque(
+            [
+                value(active_blks[t].fs.h2_tank.tank_holdup[0])
+                for t in range(last_implemented_time_step + 1)
+            ]
+        )
+
+        return {
+            "implemented_tank_holdup": implemented_tank_holdup
+        }
+
+    def record_results(self, blk, date=None, hour=None, **kwargs):
+
+        """
+        Record the operations stats for the model.
+
+        Args:
+            blk:  pyomo block
+            date: current simulation date
+            hour: current simulation hour
+
+        Returns:
+            None
+        """
+        active_blks = blk.nuclear.get_active_process_blocks()
+        df_list = []
+        for t in blk.HOUR:
+            result_dict = {
+                "Date": date,
+                "Hour": hour,
+                "Horizon [hr]": int(t),
+                "Power to Grid [MW]": float(round(value(blk.P_T[t]), 2)),
+                "Power to PEM [MW]": float(round(value(active_blks[t].fs.pem.electricity[0]) * 1e-3, 2)),
+                "Initial holdup [kg]": float(round(value(active_blks[t].fs.h2_tank.tank_holdup_previous[0])
+                                                   * 2.016 * 1e-3, 2)),
+                "Final holdup [kg]": float(round(value(active_blks[t].fs.h2_tank.tank_holdup[0])
+                                                 * 2.016 * 1e-3, 2)),
+                "Hydrogen Market [kg/hr]": float(round(value(active_blks[t].fs.h2_tank.outlet_to_pipeline.flow_mol[0])
+                                                       * 2.016 * 1e-3 * 3600, 2)),
+                "Total Cost [$]": float(round(value(blk.tot_cost[t]), 2))}
+
+            for key in kwargs:
+                result_dict[key] = kwargs[key]
+
+            result_df = pd.DataFrame.from_dict(result_dict, orient="index")
+            df_list.append(result_df.T)
+
+        # append to result list
+        self.result_list.append(pd.concat(df_list))
+
+        return
+
+    def write_results(self, path):
+
+        """
+        Write the saved results to a csv file.
+        
+        Args:
+            path: the path to write the results.
+        Return:
+            None
+        """
+
+        pd.concat(self.result_list).to_csv(path, index=False)
+
+    @property
+    def power_output(self):
+        return "P_T"
+
+    @property
+    def total_cost(self):
+        return ("tot_cost", 1)
+
+    # @property
```

Comment:
If these will not be used in future, then may be consider removing these lines

Reply:
Removed the lines from the file.